Feat/add certification process2 (#781)
SovereignCloudStack/standards: Standards/scs-0004-v1-achieving-certification.md

## Regulations

> 0. Certificates are issued by the SCS certification assessment body (initially the SCS project in the OSB Alliance e.V., to be succeeded by the Forum SCS-Standards in the very same OSB Alliance). An interested party has to apply for certification with this body, which in turn determines the rules that govern what parties are eligible for application (fees may apply).

Reviewer: ```suggestion
0. Certificates are issued by the SCS certification assessment body. This is either the _Forum SCS-Standards_, part of the Open Source Business Alliance e.V. (OSBA), or an organization appointed by the _Forum SCS-Standards_ . An interested party has to apply for certification with this body, which in turn determines the rules that govern what parties are eligible for application (fees may apply).
```

> 1. Each certificate issued pertains to a given combination of subject (i.e., cloud environment), scope (such as _SCS-compatible IaaS_, and version of that scope. The certificate is only valid for that combination and for the time frame that ends when the scope expires, or for six months if the expiration date for the scope is not yet fixed.

Reviewer 2: I guess there is a bracket mnissing 
```suggestion
1. Each certificate issued pertains to a given combination of subject (i.e., cloud environment), scope (such as _SCS-compatible IaaS_), and version of that scope. The certificate is only valid for that combination and for the time frame that ends when the scope expires, or for six months if the expiration date for the scope is not yet fixed.
```

   For public clouds, the SCS certification assessment body can take on this task provided that suitable access to test subject is supplied.

   The test suite is partitioned according to resource usage; the required test intervals depend on this classification:

    - _light_: at least nightly,
    - _medium_: at least weekly,
    - _heavy_: at least monthly.

3. If the desired certificate requires manual checks, then the operator MUST offer the SCS project suitable documentation. Manual checks MUST be repeated once every quarter. In addition, the SCS certification assessment body reserves the right to occasionally verify documentation on premises.

4. Details on the standards achieved, as well as the current state and the history of all test and check results of the past 18 months will be displayed on a public webpage (henceforth, _certificate status page_) owned by SCS.

>    The page will be kept online for the duration of the certificate's validity, plus at least 3 months; afterwards, it can be taken offline, either upon request or in the course of maintenance cleanup. However, the page's content won't be deleted until 12 months after the certificate's expiration, for the page will be reanimated and reused if, within this timeframe, a new certificate is issued for the same scope and the same cloud.

Reviewer 2: IMO, the deletion time is not clear, but could be very important for providers, especially, if test suite failed frequently for a specific certificate scope and version. 

> the page will be reanimated and reused if, within this timeframe, a new certificate is issued for the same scope and the same cloud.

Does this means, that the entire history of all test results for all previous certificate version is refreshed in case a new version is issued? E.g. SCS-compatible IaaS v1.0, SCS-compatible IaaS v2.0, SCS-compatible IaaS v3.0, SCS-compatible IaaS v4.0 and SCS-compatible IaaS v5.0 is still visible in Compliance Monitor, in case SCS-compatible IaaS v6.0 is issued?

IMO, we should be more precise in deletion time, e.g. "... the last three version (if present) will be visible, in case a new certificate version is issued. Test results older then three past version will be deleted 3 month after expiring date.

Reviewer: Discussed with @mbuechse: This is a valid point. We want to defer the discussion to a later step in the process.

> 5. The SCS certification assessment body WILL review the certification application and either grant the certification, reject it or ask for further measures or information.

Reviewer: ```suggestion
5. The SCS certification assessment body, represented by the Forum SCS-Standards of the OSBA, WILL review the certification application and either grant the certification, reject it or ask for further measures or information.
```

Reviewer: not needed, since this is defined in the first paragraph

6. Once the certificate is granted by the SCS certification assessment body, the operator SHOULD use the corresponding logo and publicly state the certified "SCS compatibility" on the respective layer for the time of the validity of the certification. In case of a public cloud, this public display is even REQUIRED. In any case, the logo MUST be accompanied by a hyperlink (a QR code for printed assets) to the respective certificate status page.

> 7. If the certificate is to be revoked for any reason, it will be included in a publicly available Certificate Revocation List (CRL). This fact will also be reflected in the certificate status page.

Reviewer 2: Who is maintaining this list. SCS Forum?

> 7. If the certificate is to be revoked for any reason, it will be included in a publicly available Certificate Revocation List (CRL). This fact will also be reflected in the certificate status page.

Reviewer: ```suggestion
7. If the certificate is to be revoked for any reason, it will be included in a publicly available Certificate Revocation List (CRL), maintained by the SCS certification assessment body. This fact will also be reflected in the certificate status page.
```

> 8. If any of the automated tests or manual checks fail after the certificate has been issued, the certificate is not immediately revoked. Rather, the automated tests MUST pass 99.x % of the runs, and the operator SHALL be notified at the second failed attempt in a row at the latest. In case a manual check fails, it has to be repeated at a date to be negotiated with the SCS certification assessment body. It MAY NOT fail more than two times in a row.

Reviewer 2: How does the information look like? E-Mail? Maybe we should say something about the requirements for this information. e.g. 

"... provider MUST provide a valid E-Mail to be used for information about failed test suite"

Reviewer: not relevant for the moment being
